scripts/languages.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs')
+const _ = require('lodash')
+
+const LinguistLanguages = require('linguist-languages')
+
+/**
+ *
+ * @param {LinguistLanguages.LinguistLanguages} languages
+ * @param {'dot-properties'} parser
+ * @param {string[]} aceModes
+ * @returns
+ */
+const getSupportLanguages = (languages, parser, aceModes) =>
+  Object.entries(languages).reduce((acc, [name, language]) => {
+    const {
+      aceMode,
+      tmScope,
+      codemirrorMode,
+      codemirrorMimeType,
+      languageId: linguistLanguageId
+    } = /** @type {LinguistLanguages.Language} */ (language)
+    if (!aceModes.includes(aceMode)) {
+      return acc
+    }
+    acc.push({
+      name,
+      since: '0.1.0',
+      parsers: [parser],
+      ..._.pick(
+        language,
+        'group',
+        'aliases',
+        'extensions',
+        'filenames',
+        'interpreters'
+      ),
+      tmScope,
+      aceMode,
+      codemirrorMode,
+      codemirrorMimeType,
+      linguistLanguageId,
+      vscodeLanguageIds: [aceMode]
+    })
+    return acc
+  }, [])
+fs.writeFileSync(
+  'languages.js',
+  `exports.languages = ${JSON.stringify(
+    getSupportLanguages(LinguistLanguages, 'dot-properties', ['properties']),
```

Comment:
```suggestion
    LinguistLanguages['Java Properties'],
```

This doesn't need to be that complicated. Either we should just write the contents of the entry in `linguist-languages`, or simply build something like the `language` value that this PR removes from `index.js`. There's no need for `reduce()`, or for `_.pick()` in particular.

Reply:
`properties` is not only bound with `['Java Properties']`.

```js
exports.languages = [
  {
    "name": "Java Properties",
    "since": "0.1.0",
    "parsers": [
      "dot-properties"
    ],
    "extensions": [
      ".properties"
    ],
    "tmScope": "source.java-properties",
    "aceMode": "properties",
    "codemirrorMode": "properties",
    "codemirrorMimeType": "text/x-properties",
    "linguistLanguageId": 519377561,
    "vscodeLanguageIds": [
      "properties"
    ]
  },
  {
    "name": "TextMate Properties",
    "since": "0.1.0",
    "parsers": [
      "dot-properties"
    ],
    "aliases": [
      "tm-properties"
    ],
    "filenames": [
      ".tm_properties"
    ],
    "tmScope": "source.tm-properties",
    "aceMode": "properties",
    "codemirrorMode": "properties",
    "codemirrorMimeType": "text/x-properties",
    "linguistLanguageId": 981795023,
    "vscodeLanguageIds": [
      "properties"
    ]
  }
]
```

Reply:
`.tm_properties` is a different file format, which is not a subset of `.properties`. It includes e.g. `[ foo.bar ]` section headers. Trying to use this plugin with them is likely to produce surprising results.

Reply:
Then it is strange `codemirror` says they're same?

Reply:
It looks like CodeMirror's "properties" mode handles an inclusive set of features of the two formats: https://github.com/codemirror/legacy-modes/blob/main/mode/properties.js

The `dot-properties` package used here does not, however, as it only supports the Java properties syntax.

Reply:
Thanks for pointing out. I'll change this PR then.

Maybe we can add support for `. tm_properties` in `dot-properties`?

Reply:
@eemeli Done!

Reply:
> Maybe we can add support for `. tm_properties` in `dot-properties`?

That's rather unlikely, tbh. As far as I can tell, the format is only used for TextMate's own settings files, and its grammar is rather different from the Java properties grammar: https://github.com/textmate/textmate/blob/master/Applications/TextMate/resources/TextMate%20Help/properties.md#grammar

Reply:
Never mind whether we're unlikely to support it.